cparser/Elab.ml

```diff
@@ -2304,7 +2363,8 @@ let rec elab_stmt env ctx s =
   | If(a, s1, s2, loc) ->
       let a',env = elab_expr ctx.ctx_vararg loc env a in
       if not (is_scalar_type env a'.etyp) then
-        error loc "the condition of 'if' does not have scalar type";
+        error loc "statement requires expression of scalar type (%a invalid)"
```

Comment:
Best of both messages: `the condition of 'if' does not have scalar type (%a invalid)`